File: Corriges/Ex0/files0.exs
defmodule FileFinder do
  def find_all_file_paths_starting_with(prefix, path \\ ".") do
    File.ls!(path)
    |> Enum.map(&Path.join(path, &1))
    |> Enum.flat_map(
      fn(pathname) ->
        cond do
          File.dir?(pathname) -> find_all_file_paths_starting_with(prefix, pathname)
          String.starts_with?(Path.basename(pathname), prefix) -> [pathname]
          true -> []
        end
      end)
  end
end

IO.puts FileFinder.find_all_file_paths_starting_with(hd(System.argv), Enum.at(System.argv, 1)) |> Enum.join("\n")

File: Corriges/Ex0/files0.rb

def find_all_file_paths_starting_with(starts, path)
  files = []
  Dir.glob(File.join(path, '*')).each do |item|
    if File.file?(item) && File.basename(item).start_with?(starts)
      files << File.expand_path(item)
    elsif File.directory?(item)
      files.concat(find_all_file_paths_starting_with(starts, item))
    end
  end
  files
end

puts find_all_file_paths_starting_with(ARGV[0], ARGV[1])
